lib/puppet/type.rb

```diff
@@ -1909,9 +1909,7 @@ def self.doc
         # Since we're mixing @doc with text from other sources, we must normalize
         # its indentation with scrub. But we don't need to manually scrub the
         # provider's doc string, since markdown_definitionlist sanitizes its inputs.
-        scrub(@doc) + "Available providers are:\n\n" + parenttype.providers.sort { |a,b|
-          a.to_s <=> b.to_s
-        }.collect { |i|
+        scrub(@doc) + "Available providers are:\n\n" + parenttype.providers.sort_by(&:to_s).collect { |i|
```

Comment:
This should be wrapped in a `do..end` instead of `{..}` because the block spwans on multiple lines

Reply:
There's a https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/BlockDelimiters cop for that, but it's out of scope for this PR.